internal/service/alertconfiguration/model_alert_configuration.go

```diff
@@ -0,0 +1,333 @@
+		Matcher:               NewTFMatcherModelList(apiRespConfig.Matchers, currState.Matcher),
+	}
+}
+
+func NewTFNotificationModelList(n []admin.AlertsNotificationRootForGroup, currStateNotifications []TfNotificationModel) []TfNotificationModel {
+	notifications := make([]TfNotificationModel, len(n))
+
+	if len(n) != len(currStateNotifications) { // notifications were modified elsewhere from terraform, or import statement is being called
+		for i := range n {
+			value := n[i]
+			notifications[i] = TfNotificationModel{
+				TeamName:       conversion.StringPtrNullIfEmpty(value.TeamName),
+				Roles:          value.Roles,
+				ChannelName:    conversion.StringPtrNullIfEmpty(value.ChannelName),
+				DatadogRegion:  conversion.StringPtrNullIfEmpty(value.DatadogRegion),
+				DelayMin:       types.Int64PointerValue(conversion.IntPtrToInt64Ptr(value.DelayMin)),
+				EmailAddress:   conversion.StringPtrNullIfEmpty(value.EmailAddress),
+				IntervalMin:    types.Int64PointerValue(conversion.IntPtrToInt64Ptr(value.IntervalMin)),
+				MobileNumber:   conversion.StringPtrNullIfEmpty(value.MobileNumber),
+				OpsGenieRegion: conversion.StringPtrNullIfEmpty(value.OpsGenieRegion),
+				TeamID:         conversion.StringPtrNullIfEmpty(value.TeamId),
+				NotifierID:     types.StringPointerValue(value.NotifierId),
+				TypeName:       conversion.StringPtrNullIfEmpty(value.TypeName),
+				Username:       conversion.StringPtrNullIfEmpty(value.Username),
+				EmailEnabled:   types.BoolValue(value.EmailEnabled != nil && *value.EmailEnabled),
+				SMSEnabled:     types.BoolValue(value.SmsEnabled != nil && *value.SmsEnabled),
+			}
+		}
+		return notifications
+	}
+
+	for i := range n {
+		value := n[i]
+		currState := currStateNotifications[i]
+		newState := TfNotificationModel{
+			TeamName: conversion.StringPtrNullIfEmpty(value.TeamName),
+			Roles:    value.Roles,
+		}
+
+		// sentive attributes do not use value returned from API
+		newState.APIToken = conversion.StringNullIfEmpty(currState.APIToken.ValueString())
+		newState.DatadogAPIKey = conversion.StringNullIfEmpty(currState.DatadogAPIKey.ValueString())
+		newState.OpsGenieAPIKey = conversion.StringNullIfEmpty(currState.OpsGenieAPIKey.ValueString())
+		newState.ServiceKey = conversion.StringNullIfEmpty(currState.ServiceKey.ValueString())
+		newState.VictorOpsAPIKey = conversion.StringNullIfEmpty(currState.VictorOpsAPIKey.ValueString())
+		newState.VictorOpsRoutingKey = conversion.StringNullIfEmpty(currState.VictorOpsRoutingKey.ValueString())
+		newState.WebhookURL = conversion.StringNullIfEmpty(currState.WebhookURL.ValueString())
+		newState.WebhookSecret = conversion.StringNullIfEmpty(currState.WebhookSecret.ValueString())
+		newState.MicrosoftTeamsWebhookURL = conversion.StringNullIfEmpty(currState.MicrosoftTeamsWebhookURL.ValueString())
+
+		// for optional attributes that are not computed we must check if they were previously defined in state
+		if !currState.ChannelName.IsNull() {
```

Comment:
no need to check if in these stamenents, it'll have the zero value anyway if the assigned value is nil

Reply:
double checked and when entering the if it sets the value using another variable instead of `currState`. It seems that it has to be like this to keep the same behaviour.

Reply:
changed the test to test this behaviour